### PR TITLE
Add mail-feature. Now when user runs commmands like submit, runtest or paste, program will start polling new reviews from tmc-server.

### PR DESCRIPTION
**Pull request by samutamm**

### CODE REVIEW

File: config.properties [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#Updated properties
+#Thu Jun 25 10:27:41 EEST 2015
+serverAddress=https\://tmc.mooc.fi/staging
+serverPort=46461
```
Comment (Coolnesss) [inline config.properties]: Get rid of this

File: config.properties.save [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#Wed Jun 24 14:49:55 EEST 2015
+serverAddress=https\://tmc.mooc.fi/staging
+serverPort=39537
```
Comment (Coolnesss) [inline config.properties.save]: Get rid of this

File: src/main/java/hy/tmc/cli/Main.java [hunk 3]
```diff
@@ -1,15 +1,20 @@
...
 
+import hy.tmc.cli.backend.Mailbox;
+import hy.tmc.cli.domain.Review;
 import hy.tmc.cli.frontend.FrontendListener;
 import hy.tmc.cli.frontend.communication.server.ProtocolException;
 import hy.tmc.cli.frontend.communication.server.Server;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/Main.java]: why

File: src/main/java/hy/tmc/cli/backend/MailFormatter.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+     * @return formatted string which contains reviews
+     */
+    public static String formatReviews(List<Review> reviews) {
+        return reviewOutput(reviews);
+    }
+
+    /**
+     *
+     * @param exercises
+     * @return
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/backend/MailFormatter.java]: Missing javadocs

File: src/main/java/hy/tmc/cli/backend/communication/StatusPoller.java [hunk 5]
```diff
@@ -1,19 +1,69 @@
...
+     * @param schedule object that contains the time interval
+     */
+    public StatusPoller(Course currentCourse, PollScheduler schedule) {
+        this.currentCourse = currentCourse;
+        this.pollScheduler = schedule;
+    }
+
     @Override
-    protected void runOneIteration() throws Exception {
-        System.out.println("Jee from cron-like");
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/backend/communication/StatusPoller.java]: Get rid of this

File: src/main/java/hy/tmc/cli/backend/communication/UrlCommunicator.java [hunk 6]
```diff
@@ -62,14 +70,42 @@ private static void addFileToRequest(ContentBody fileBody, HttpPost httppost) {
...
+                String value = body.get().get(key);
+                params.add(new BasicNameValuePair(key, value));
+            }
+            
+            httpPut.setEntity(new UrlEncodedFormEntity(params));
+            
+            return getResponseResult(httpPut);
+        } catch (IOException e) {
+            System.err.println(e.getMessage());
+            return new HttpResult("", BAD_REQUEST, false);
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/backend/communication/UrlCommunicator.java]: This should be changed to just throw IOException

File: src/main/java/hy/tmc/cli/backend/communication/UrlCommunicator.java [hunk 7]
```diff
@@ -94,12 +130,11 @@ public static boolean downloadFile(
             HttpResponse response = executeRequest(httpget);
             fileOutputStream.write(EntityUtils.toByteArray(response.getEntity()));
             return true;
-        }
-        catch (IOException e) {
+        } catch (IOException e) {
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/backend/communication/UrlCommunicator.java]: Get rid of this

File: src/main/java/hy/tmc/cli/zipping/ProjectRootFinder.java [hunk 8]
```diff
@@ -52,6 +54,11 @@ public ProjectRootFinder(ProjectRootDetector detector) {
     @Override
     public Optional<Course> getCurrentCourse(String path) throws IOException {
         String[] foldersOfPwd = path.split("/");
```
Comment (Coolnesss) [inline src/main/java/hy/tmc/cli/zipping/ProjectRootFinder.java]: File.separator

File: src/test/java/feature/runtests/TmcTestsSteps.java [hunk 9]
```diff
@@ -54,11 +64,20 @@ public void theUserSeesWhichTestsHaveFailed() {
      * User should get both information about passed tests and failed tests.
      */
     @Then("^the user sees both passed and failed tests$")
-    public void theUserSeesBothPassedAndFailedTests() {
+    public void theUserSeesBothPassedAndFailedTests() throws UnsupportedEncodingException {
```
Comment (Coolnesss) [inline src/test/java/feature/runtests/TmcTestsSteps.java]: Get rid of this

File: src/test/java/hy/tmc/cli/backend/CacheTest.java [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+import java.util.HashMap;
+import static org.junit.Assert.assertNotEquals;
+import org.junit.Test;
+
+public class CacheTest {
+
+    @Test
+    public void timestampTest() throws InterruptedException {
+        Date date = new Date();
+        Thread.sleep(10);
```
Comment (Coolnesss) [inline src/test/java/hy/tmc/cli/backend/CacheTest.java]: Get rid of this

File: src/test/java/hy/tmc/cli/frontend/communication/commands/ListCoursesTest.java [hunk 11]
```diff
@@ -78,7 +78,7 @@ public void testWithAuthPrintsSeveralCourses() throws Exception {
             String testResult = list.parseData(list.call()).get();
             assertTrue(testResult.contains("WEPATEST"));
         } catch (ProtocolException ex) {
-            fail("unexpected exception");
```
Comment (Coolnesss) [inline src/test/java/hy/tmc/cli/frontend/communication/commands/ListCoursesTest.java]: Should just throw exception, no catch and fail

File: src/test/java/hy/tmc/cli/zipping/ProjectRootFinderTest.java [hunk 12]
```diff
@@ -39,7 +40,8 @@ public void setUp() throws IOException {
 
         List<Course> courses = setupFakeCourses();
         PowerMockito
-                .when(TmcJsonParser.getCourses())
+                .when(TmcJsonParser.getCourses(new ConfigHandler()
```
Comment (Coolnesss) [inline src/test/java/hy/tmc/cli/zipping/ProjectRootFinderTest.java]: I bet this doesnt work - no eq

File: testResources/failingExercise/viikko1/Viikko1_001.Nimi/build/test/results/TEST-NimiTest.xml [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+  </properties>
+  <testcase classname="NimiTest" name="test" time="0.025">
+    <failure message="Et tulostanut mitään!" type="junit.framework.AssertionFailedError">junit.framework.AssertionFailedError: Et tulostanut mitään!
+	at NimiTest.test(NimiTest.java:17)
+	at fi.helsinki.cs.tmc.edutestutils.MockStdio$1.evaluate(MockStdio.java:106)
+</failure>
+  </testcase>
+  <system-out><![CDATA[]]></system-out>
+  <system-err><![CDATA[]]></system-err>
+</testsuite>
```
Comment (Coolnesss) [inline testResources/failingExercise/viikko1/Viikko1_001.Nimi/build/test/results/TEST-NimiTest.xml]: Get rid of this

File: testResources/halfdoneExercise/viikko1/Viikko1_004.Muuttujat/build/built-jar.properties [hunk 14]
```diff
@@ -0,0 +1,4 @@
+#Thu, 25 Jun 2015 10:24:55 +0300
+
+
+/media/ExternalSD/Other/OhTu/tmc-cli/testResources/halfdoneExercise/viikko1/Viikko1_004.Muuttujat=
```
Comment (Coolnesss) [inline testResources/halfdoneExercise/viikko1/Viikko1_004.Muuttujat/build/built-jar.properties]: Get rid of this

File: testResources/successExercise/viikko1/Viikko1_001.Nimi/ignore_this.txt [hunk 15]
```diff
@@ -1,4 +0,0 @@
-# this file should not be edited
-
-<(^)
- (__)
```
Comment (Coolnesss) [inline testResources/successExercise/viikko1/Viikko1_001.Nimi/ignore_this.txt]: Maybe this should not be removed?

File: testResources/successExercise/viikko1/Viikko1_001.Nimi/test/.tmcproject.yml [hunk 16]
```diff
@@ -1,3 +0,0 @@
-extra_student_files:
-    - test/StudentTest.java
-    - duck/duck.py
```
Comment (Coolnesss) [inline testResources/successExercise/viikko1/Viikko1_001.Nimi/test/.tmcproject.yml]: why remove

File: config.properties.save (hunk 2 above)
Comment (samutamm) [inline config.properties.save]: done

**Pull request merged**